xds/src/main/java/io/grpc/xds/XdsCredentialsRegistry.java

```diff
@@ -133,7 +133,7 @@ public static synchronized XdsCredentialsRegistry getDefaultRegistry() {
    * XdsCredsProvider of that scheme. 
    */
   @VisibleForTesting
-  synchronized Map<String, XdsCredentialsProvider> providers() {
+  synchronized ImmutableMap<String, XdsCredentialsProvider> providers() {
```

Comment:
Leave this as Map. We do not expose Guava in our API. Yes, this is internal for the moment, but it is intended to be eventually public.

Reply:
Ack, reverted this back to Map.